ui/js/search.js

```diff
@@ -0,0 +1,171 @@
+var SEARCH_COURSE_API = 'http://localhost:3000/search-course?searchQuery=';
+var COURSE_MATCHER = new RegExp("^[a-zA-Z]{3}[0-9]{1,3}$");
+var chosenCourses = [];
+
+$(document).ready(function () {
+    var input = '';
+
+    String.prototype.replaceAll = function (search, replacement) {
+        var target = this;
+        return target.replace(new RegExp(search, 'g'), replacement);
+    };
+
+    $('#courseSearch').on('input', function (e) {
+        input = this.value;
+        if (input && input.length > 3 && COURSE_MATCHER.test(input)) {
+            var URL = SEARCH_COURSE_API + input;
+            $.get(URL, function (response) {
+                displayDropdown(response);
+            });
+            return true;
+        }
+        return false;
+    });
+
+
+
+});
+
+function displayDropdown(dataArr) {
+    var currentlySelected = -1;
+    var numOptions = 0;
+    var dropdownOpen = false;
+    var disableArrowSelection = false;
+
+    initDialog(dataArr);
+
+    function initDialog(courses) {
+        numOptions = courses.length;
+        $('.dialog').empty();
+        for (var course of courses) {
+            if (chosenCourses.indexOf(course.id) < 0) {
+                $('.dialog').append('<div data-course-id=' + course.id + '>' +
+                    course.courseFullName.replaceAll(":", "") + '</div>');
+            }
+        }
+        $('.dialog').addClass('open');
+        dropdownOpen = true;
+        resetEventListeners(); 
+    }
+
+    function trackArrowKeysAndEnterBtn() {
```

Comment:
Do you think it'd be a good idea to put this function on top of the `setClickListeners` function? I think it'll be better organized that way